src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponentSpec.scala

```diff
@@ -114,6 +114,39 @@ class SubmissionComponentSpec extends TestDriverComponentWithFlatSpecAndMatchers
     }
   }
 
+  it should "create multiple unique submissions concurrently" in withDefaultTestDatabase {
+    withWorkspaceContext(testData.workspace) { context =>
+      val origCount = runAndWait(submissionQuery.list(context)).length
+
+      def submissionGenerator(dummy: Int) = {
+        val uniqueSubmission = createTestSubmission(testData.workspace, testData.methodConfig2, testData.indiv1, testData.userOwner, Seq(testData.sample1, testData.sample2, testData.sample3), Map(testData.sample1 -> testData.inputResolutions, testData.sample2 -> testData.inputResolutions, testData.sample3 -> testData.inputResolutions))
+        submissionQuery.create(context, uniqueSubmission)
+      }
+
+      val count = 100
+      runMultipleAndWait(count)(submissionGenerator)
+      assertResult(origCount) {
+        runAndWait(submissionQuery.list(context)).length - count
+      }
+    }
+  }
+
+  // fails
+  it should "not create the same submission multiple times when run concurrently" in withDefaultTestDatabase {
```

Comment:
We should probably have tests in the API spec that ensure that we don't have any issues making multiple submissions concurrently (since each should be unique), if we ensure that always gives us unique submissions (unique ids), then the case below with reusing the same submission id should never occur